### ISSUE TITLE
code 4012

### ISSUE THREAD

**Issue by haxxus** [gordon1210/DiscordCoinmarketcap#3]
hello I'm getting this error:

bot disconnected with code  4012  for reason  undefined

**Comment by haxxus** [comment]
can I get some help upgrading package-lock.json .

**Comment by mcdope** [comment]
Considering the actual failed command, this seems to be an issue with your git and not the code. I got no warning, running on Ubuntu with default npm and git from the repos.

**Comment by haxxus** [comment]
gotcha , question if I wanted to not have a command rather have it pick up when the name is entered how would I accomplish this ? I believe it would be done here:

bot.on('message', function (user, userID, channelID, message, event) {
    command = message.toLowerCase().split(' ');
    if (command[0] === "$") {
        ticker = command.slice(1, command.length);
        sendCoinData(channelID, ticker);
    }
}); 

in other words ive set it to $ for now but i'm trying to not have any command at al so i can use btc instead of $btc

**Comment by mcdope** [comment]
Just check `message` contents, basic string handling - see the JavaScript docs. 

But this is a totally other topic then this issue. And not even really related to the project. Better place to ask that would be some programming forum.

**Comment by haxxus** [comment]
ok , and about my last question ? sorry i edited the text maybe you didn't see it

**Comment by mcdope** [comment]
I don't want to be rude, or ride on rules/standards, but that also doesn't belong here at all. This issue is (or was) about code 4012. You're just cluttering it, imagine you would google the code - get here, and read mainly stuff which is totally offtopic...

For the sake of politeness I will answer anyway.

I'm not sure that possible, but if it's - it needs to be done combined in `call()` and `sendCoinData()`. Just take a look at the code, this isn't rocket science - just JS. But I wouldn't wonder if it's impossible.

**Comment by haxxus** [comment]
for the sake of thanks , thanks.
